src/include/duckdb/common/preserved_error.hpp

```diff
@@ -21,6 +21,17 @@ class PreservedError {
 	PreservedError(const std::exception &ex)
 	    : initialized(true), type(ExceptionType::INVALID), raw_message(SanitizeErrorMessage(ex.what())),
 	      exception_instance(nullptr) {
+		// Given a message in the form: 	xxxxx Error: yyyyy
+		// Try to match xxxxxxx with known error so to potentially reconstruct the original error type
+		string err = raw_message.substr(0, raw_message.find(':'));
```

Comment:
I think one problem with this is the assumption that this is always an error produced by duckdb, I think we can also catch regular c++ exceptions with this, so maybe we should first do a pass to check that this is one our exceptions

Reply:
I think this is the role of line 28, where we are checking a few conditions that should be equivalent to "there is a substring ` Error:`, and the first semicolon is part of that substring.

Regular std::exceptions might also be wrongly catched iff they follow the pattern and also `err` matches one of the known errors, but that should be acceptable (here we are duck-typing exceptions 🦆, pun intended)

Reply:
If you want to review line 26-28, it would be welcome, it's likely the weaker point of the PR.

Reply:
Agreed with Thijs that this looks like it can go wrong. Can we check if `raw_message.find(':')` returns `std::string::npos ` and bail-out if it does? Plus a check if there is a message afterwards (so an error like this `error:` does not trigger undefined behavior)?

Reply:
Ouch, I just tested it plus read the spec, and you are both very right @Tishj and @Mytherin.